### PR TITLE
[data grid] remove unnecessary comment in `useGridApiOptionHandler`

### PR DESCRIPTION
**Pull request by k-rajat19**
doing this will ensure that we are not registering multiple event handlers with the same event name :)
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

**Review by romgrk — COMMENTED** [review]
Thanks for the work, but I don't think we should merge this :| I don't see a need to add validation code if it's not catching an actual problem.

**Comment by k-rajat19** [comment]
This will help prevent any future regressions. 
Currently, we are not registering multiple event handlers with the same event name anywhere in the codebase but in case we do so somewhere deep in the codebase and forget that we have already registered a handler for that event name, it will prevent that.
Also, we are exporting `useGridApiOptionHandler` so if users try to do the same it will prevent them too.

these are just my POV's, you know better than me. Feel free to close this if you want :)

**Comment by romgrk** [comment]
I'd rather close it, we have tests for all new features so I doubt we wouldn't catch an issue like this.

File: packages/x-data-grid/src/hooks/utils/useGridApiEventHandler.ts [hunk 1]
```diff
@@ -117,12 +117,23 @@ export const useGridApiEventHandler = createUseGridApiEventHandler(registryConta
 
 const optionsSubscriberOptions: EventListenerOptions = { isFirst: true };
 
+const registeredEvents = new Set<GridEvents>();
 export function useGridApiOptionHandler<Api extends GridApiCommon, E extends GridEvents>(
   apiRef: React.MutableRefObject<Api>,
   eventName: E,
   handler?: GridEventListener<E>,
 ) {
-  // Validate that only one per event name?
```
Comment (k-rajat19) [inline packages/x-data-grid/src/hooks/utils/useGridApiEventHandler.ts:125]: @romgrk  Does it make sense to remove this comment, which motivates me to add these changes 😅

Comment (romgrk) [reply packages/x-data-grid/src/hooks/utils/useGridApiEventHandler.ts:125]: Sure, that's an old comment. If you want you can update this PR directly, I'll re-open & merge.

Comment (k-rajat19) [reply packages/x-data-grid/src/hooks/utils/useGridApiEventHandler.ts:125]: I have pushed a commit in this PR, could you re-open it?

**Comment by k-rajat19** [comment]
i think this needs your approval.

**Pull request merged**